VultisigApp/VultisigApp/Services/BlockChainService.swift

```diff
@@ -92,25 +94,36 @@ final class BlockChainService {
             sendMaxAmount: false,
             isDeposit: tx.isDeposit,
             transactionType: .unspecified, 
-            gasLimit: nil,
+            gasLimit: nil, 
+            byteFee: nil,
             fromAddress: nil,
             toAddress: nil,
             feeMode: .fast
         )
 
         return specific
     }
+
+    func fetchUTXOFee(coin: Coin, action: Action, feeMode: FeeMode) async throws -> BigInt {
+        let sats = try await utxo.fetchSatsPrice(coin: coin)
+        let normalized = Self.normalizeUTXOFee(sats, action: action)
+        let prioritized = Float(normalized) * feeMode.utxoMultiplier
+        return BigInt(prioritized)
+    }
 }
 
 private extension BlockChainService {
 
-    func fetchSpecific(for coin: Coin, action: Action, sendMaxAmount: Bool, isDeposit: Bool, transactionType: VSTransactionType, gasLimit: BigInt?, fromAddress: String?, toAddress: String?, feeMode: FeeMode) async throws -> BlockChainSpecific {
+    func fetchSpecific(for coin: Coin, action: Action, sendMaxAmount: Bool, isDeposit: Bool, transactionType: VSTransactionType, gasLimit: BigInt?, byteFee: BigInt?, fromAddress: String?, toAddress: String?, feeMode: FeeMode) async throws -> BlockChainSpecific {
         switch coin.chain {
         case .bitcoin, .bitcoinCash, .litecoin, .dogecoin, .dash:
-            let sats = try await utxo.fetchSatsPrice(coin: coin)
-            let normalized = Self.normalizeUTXOFee(sats, action: action)
-            return .UTXO(byteFee: normalized, sendMaxAmount: sendMaxAmount)
-
+            let byteFeeValue: BigInt
+            if let byteFee {
```

Comment:
Here , if user just click "Send" button ,  we should proceed to `fetchUTXOFee`, otherwise it shows like this
<img width="1236" alt="image" src="https://github.com/user-attachments/assets/b8ab0ddf-6d47-4103-bc08-f3a618ed2d9c">
Noticed that "Gas (auto)" , is 0 BTC/vbyte